manifests/linux.pp

```diff
@@ -44,7 +44,13 @@
     'stopped' => false,
   }
 
-  package { 'iptables':
+  if $facts['os']['name'] == 'Fedora' and versioncmp($facts['os']['release']['full'], '34') >= 0 {
+    $_iptables_pkg = 'iptables-compat'
```

Comment:
maybe we should set this in the params.pp? that currently holds the logic to determine data based on the OS

Reply:
Actually: I am still a bit puzzled why (in the original code) there is a $package_name-variable defined in params.pp, but still the package "iptables" is required hardcoded in linux.pp then.

Reply:
I have worked around by doing the following:
```
  ## This is a bodge!! Fedora 34 onwards has no package iptables, its a meta package and breaks the parsing.
  if (($::operatingsystem == 'Fedora' and versioncmp($::operatingsystemrelease, '34') >= 0)) {
    package { 'iptables':
      name   => 'iptables-compat',
      ensure => $pkg_ensure,
    }
  } else {
    package { 'iptables':
      ensure => $pkg_ensure,
    }
  }
```

Reply:
However, maybe as @sircubbi says, this should be a param but like this:

```
if this {
$pkg_name = "iptables"
} else {
$pkg_name = "iptables-compat"
}

package { 'iptables':
 name => $pkg_name,
 ensure => $pkg_ensure,
}
```

As this would allow the code to be dropped in but also support different OS package names better in the future.